core/iwasm/include/wasm_export.h

```diff
@@ -1436,6 +1477,21 @@ wasm_runtime_get_export_global_inst(const wasm_module_inst_t module_inst,
                                     const char *name,
                                     wasm_global_inst_t *global_inst);
 
+/**
+ * Get an export memory instance
+ *
+ * @param module_inst the module instance
+ * @param name the export memory name
+ * @param memory_inst location to store the memory instance
+ *
+ * @return true if success, false otherwise
+ *
+ */
+WASM_RUNTIME_API_EXTERN bool
+wasm_runtime_get_export_memory_inst(const wasm_module_inst_t module_inst,
+                                    const char *name,
+                                    wasm_memory_inst_t *memory_inst);
```

Comment:
Don't need to export this API to get memory instance's base address and size, use:
```C
WASM_RUNTIME_API_EXTERN void *
wasm_runtime_addr_app_to_native(wasm_module_inst_t module_inst,
                                uint64_t app_offset);

WASM_RUNTIME_API_EXTERN bool
wasm_runtime_get_native_addr_range(wasm_module_inst_t module_inst,
                                   uint8_t *native_ptr,
                                   uint8_t **p_native_start_addr,
                                   uint8_t **p_native_end_addr);
```
instead, e.g.:
```C
uint8 *mem_begin_addr, *mem_end_addr;

mem_begin_addr = wasm_runtime_addr_app_to_native(module_inst, 0);

wasm_runtime_get_native_addr_range(module_inst, mem_begin_addr, NULL, &mem_end_addr);
```

Reply:
Hi @wenyongh, I think that method is sufficient currently, but potentially future WebAssembly specifications might support  multiple `memory` objects in the wasm file:

> In the current version of WebAssembly, at most one memory may be defined or imported in a single module, and all constructs implicitly reference this memory. This restriction may be lifted in future versions.
(https://webassembly.github.io/spec/core/syntax/modules.html#syntax-mem)

Is that something we want to be able to support without changing the API design again? Or would it be such a comprehensive change that we'd expect to change a number of other APIs anyway?

Reply:
The multi-memory for classic interpreter feature is under development and new APIs will be introduced:
https://github.com/bytecodealliance/wasm-micro-runtime/issues/3381

I think you can still use `wasm_runtime_addr_app_to_native_ex` and `wasm_runtime_get_native_addr_range_ex` by passing mem_idx argument to them. Had better not introduce new API if unnecessary since there are lots of APIs now.

Reply:
I understand, I have removed it.